resources/fetchersManager/factory.go

```diff
@@ -40,7 +40,7 @@ func newFactories() factories {
 	return factories{m: make(map[string]fetching.Factory)}
 }
 
-func (fa *factories) ListFetcherFactory(name string, f fetching.Factory) {
+func (fa *factories) SetFetcherFactory(name string, f fetching.Factory) {
```

Comment:
Can we change to `RegisterFetcher`/`RegisterFetcherInFactody`?

I think that if we use a known pattern we should stick with the conventions 🥇

Reply:
register implies registry and could be very confusing, we already have `RegisterFetcher` function in the registry so I think you can see why it can be confusing with our current registry. What convention?

Reply:
This is the closest example that I cloud find:

https://medium.com/@geoffreykoh/implementing-the-factory-pattern-via-dynamic-registry-and-python-decorators-479fc1537bbe

I don't mind finding the middle ground - 
`AddFetcher`, `EnlistFetcher`.

Set gives the impression of replacing changing a value from `a` to `b`